### PR TITLE
First attempt at new install

### PR DESCRIPTION
**Pull request by micke2k**
1. Added check for RRD 1.5
2. Backported dependencies check
3. Backported template installer
4. Added theme selection

Tested and working with a new installation on Linux. Backported a template for testing as well -> Disk IO.

I will test Upgrade and Windows installation, as well as clean it up and add some extras.

### CODE REVIEW

File: install/0_8_8g_to_1_0_0.php [hunk 1]
```diff
@@ -365,8 +364,8 @@ function upgrade_to_1_0_0() {
 	// Add secpass fields
 	db_install_add_column ('1.0', 'user_auth', array('name' => 'lastchange', 'type' => 'int(12)', 'NULL' => false, 'default' => '-1'));
 	db_install_add_column ('1.0', 'user_auth', array('name' => 'lastlogin', 'type' => 'int(12)', 'NULL' => false, 'default' => '-1'));
-	db_install_add_column ('1.0', 'user_auth', array('name' => 'password_history', 'type' => 'varchar(4096)', 'NULL' => false, 'default' => ''));
-	db_install_add_column ('1.0', 'user_auth', array('name' => 'locked', 'type' => 'varchar(3)', 'NULL' => false, 'default' => ''));
+	db_install_add_column ('1.0', 'user_auth', array('name' => 'password_history', 'type' => 'INT(12)', 'NULL' => false, 'default' => '-1'));
```
Comment (cigamit) [inline install/0_8_8g_to_1_0_0.php]: mistake here.  should remain varchar(4096).  avoid use of 'text' fields at all cost.

File: install/0_8_8g_to_1_0_0.php [hunk 2]
```diff
@@ -365,8 +364,8 @@ function upgrade_to_1_0_0() {
...
+	db_install_add_column ('1.0', 'user_auth', array('name' => 'locked', 'type' => 'text', 'NULL' => false, 'default' => ''));
```
Comment (cigamit) [inline install/0_8_8g_to_1_0_0.php]: should remain varchar(3)

File: install/0_8_8g_to_1_0_0.php [hunk 3]
```diff
@@ -464,13 +463,13 @@ function upgrade_to_1_0_0() {
...
 	db_install_add_column ('1.0', 'user_auth', array('name' => 'email_address', 'type' => 'varchar(128)', 'NULL' => true, 'after' => 'full_name'));
-	db_install_add_column ('1.0', 'user_auth', array('name' => 'password_change', 'type' => 'char(2)', 'NULL' => false, 'default' => 'on', 'after' => 'email_address'));
+	db_install_add_column ('1.0', 'user_auth', array('name' => 'password_change', 'type' => 'char(2)', 'NULL' => true, 'default' => 'on', 'after' => 'must_change_password'));
 
 	db_install_execute('1.0', 'DROP TABLE IF EXISTS poller_output_realtime');
 	db_install_execute('1.0', "CREATE TABLE poller_output_realtime (
 		local_data_id mediumint(8) unsigned NOT NULL default '0',
 		rrd_name varchar(19) NOT NULL default '',
-		time timestamp NOT NULL default '0000-00-00 00:00:00',
+		time datetime NOT NULL default '0000-00-00 00:00:00',
```
Comment (cigamit) [inline install/0_8_8g_to_1_0_0.php]: please avoid the use of datetime.

File: install/0_8_8g_to_1_0_0.php [hunk 4]
```diff
@@ -1189,11 +1188,11 @@ function upgrade_to_1_0_0() {
 	
 	db_install_execute('1.0', "ALTER TABLE settings MODIFY COLUMN value varchar(2048) NOT NULL default ''");
 	db_install_execute('1.0', "ALTER TABLE settings_graphs MODIFY COLUMN value varchar(2048) NOT NULL default ''");
-	db_install_execute('1.0', "ALTER TABLE user_auth MODIFY COLUMN password varchar(2048) NOT NULL default ''");
+	db_install_execute('1.0', "ALTER TABLE user_auth MODIFY COLUMN password varchar(2048) NOT NULL default '0'");
```
Comment (cigamit) [inline install/0_8_8g_to_1_0_0.php]: i think default should be ''.  why '0'.  might have been some mistake.

**Comment by cigamit** [comment]
I've made some comments.  If you can update, that would be good.

**Comment by micke2k** [comment]
Ill fix it. The values comes from cacti.sql, ill fix that one too. Thanks for the comments!

**Comment by cigamit** [comment]
before merging, let's do those other two changes.

**Comment by micke2k** [comment]
yup agreed

**Comment by micke2k** [comment]
sqlcompare

**Comment by cigamit** [comment]
yea, the index should also be on the cacti.sql file.  I'll merge this.  add the parent index to the cacti.sql in a separate pull.

**Pull request merged**